### PR TITLE
Enable local partial templates that override global template

### LINKED ISSUE #1893: Allow templates to be provided in the `pkgdown/` directory
A package author might want to make limited changes to a parent pkgdown template, or perhaps provide an additional customization for a template.

Much like hugo accepts custom layout templates at the site level that take precedence over the global theme's layout, I'd like to propose that templates in a `pkgdown/templates` directory in a package's project take precedence over the templates provided by the template defined in `_pkgdown.yml` or those provided by `pkgdown` directly. The end result is that it becomes easier to make small modifications without needing the additional infrastructure of a package or without having to make upstream modifications to the template package.

If this seems reasonable, I'd be happy to provide a PR. I have a working local implementation that adds `path(pkg$src_path, "pkgdown", "templates")` to the front of the [`template_candidates()`](https://github.com/pkgdown/pkgdown/blob/3912737f46bf1f9f04ce59ee333b685d7bff599f/R/templates.R#L33) paths here:

https://github.com/r-lib/pkgdown/blob/3912737f46bf1f9f04ce59ee333b685d7bff599f/R/templates.R#L39-L42

Since `template_candidates()` doesn't take `pkg` as an argument, we'd need to update the function signature and ensure that `src_path` is passed down through [`find_template()`](https://github.com/r-lib/pkgdown/blob/3912737f46bf1f9f04ce59ee333b685d7bff599f/R/templates.R#L1-L5) from its calling functions, much like how `templates_dir` is handled currently.

### PR DESCRIPTION
**Pull request by gadenbuie**
Fixes #1893 by adding a `src_path` argument to `find_templates()` and passing `pkg$src_path` to this argument wherever `find_templates()` is called.

Two follow up questions:

1. Would you like this feature documented anywhere specific? I looked around for a good place but didn't find a good fit — perhaps at the end of [Customize - Template Packages](https://pkgdown.r-lib.org/dev/articles/customise.html#template-packages)?

1. At this point `find_templates()` requires `pkg` for 3 of its arguments. I could refactor `find_templates()` to take `pkg` as an argument, likely the first argument in line with other functions in pkgdown. The only downside I see is that we'd end up calculating `templates_dir` a few more times than necessary if we call `templates_dir(pkg)` inside `find_templates()`. I don't think that would have a noticeable impact, though. 

   I'd be happy to add the refactoring to this PR if desired.

### CODE REVIEW

File: R/templates.R [hunk 1]
```diff
@@ -2,14 +2,16 @@ find_template <- function(type,
                           name,
                           ext = ".html",
                           templates_dir = NULL,
-                          bs_version = 3) {
+                          bs_version = 3,
```
Comment (hadley) [inline R/templates.R:5]: I wonder if it's now the time to bite the bullet and pass `pkg` to this function?

Comment (gadenbuie) [reply R/templates.R:5]: I agree, I'll add that change

**Comment by hadley** [comment]
How about a new subsection at the end of https://pkgdown.r-lib.org/dev/articles/customise.html#additional-html-and-files — I think you could just briefly mention this as an option for experts. And can you please add a news bullet somewhere in https://github.com/r-lib/pkgdown/blob/HEAD/NEWS.md#customisation?

**Comment by hadley** [comment]
Thanks!

**Pull request merged**